src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -257,14 +257,26 @@ class device {
         sensor_tree::put( "board.physical.electrical.12v_aux.voltage", m12v_aux_vol ); 
         sensor_tree::put( "board.physical.electrical.12v_aux.current", m12v_aux_curr );
 
+        // U50 has different power rails, so different calculation
```

Comment:
u50 probably don't have 12v aux? if so, that should read as 0, right? so we can have common code for power calculation (just include all three rails)?

Reply:
Seems to work - see new commit